vendor/github.com/kubev2v/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -460,6 +455,28 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 		// We'll get called again later when the data source exists
 		return nil
 	}
+	var rawBlock bool
+	if nil != pvc.Spec.VolumeMode && corev1.PersistentVolumeBlock == *pvc.Spec.VolumeMode {
+		rawBlock = true
+	}
+
+	// Calculate the args for the populator pod
+	var args []string
+	args, err = c.populatorArgs(rawBlock, unstructured)
+	if err != nil {
+		return err
+	}
+
+	// Overriding the namespace to our target namespace
+	// Taking the secret name from the args
+	var secretName string
+	for _, val := range args {
+		if strings.HasPrefix(val, "--cr-namespace=") {
+			c.populatorNamespace = strings.Split(val, "--cr-namespace=")[1]
```

Comment:
why is this needed? don't we set the namespace to that of the CR?

Reply:
no, that's the problem. the `c.populatorNamespace` is set with the value given from the `populator-controller` and it's the namespace the `populator-controller` is running. with this line, we take the namespace from the `OvirtVolumePopulator`.

Reply:
ack so can't we change in the populator-controller to provide the namespace that the CR resides in?

Reply:
ok, I missed that it's not part of the populator pod but the populator controller and we still have that populatorNamespace that is initialized to POD_NAMESPACE - this requires a bit more cleanup